C/c4Internal.hh

```diff
@@ -33,6 +33,28 @@ using namespace litecore;
 #define LOCK(MUTEX)     std::unique_lock<std::mutex> _lock(MUTEX)
 #define UNLOCK()        _lock.unlock();
 
+#if defined(__clang__)
```

Comment:
What the heck is this?

Reply:
This is a way to disable compiler warnings for specific blocks of code.  In particular, Apple would complain mightily about the fact that I check for a "tautological falsehood" of the "non null" replicator being null, and then again it would (rightfully) complain about me passing in null during testing to check that it rejects it.